src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -1048,6 +1057,199 @@ bool needsReporting()
+                else
+                {
+                    // If an error needs to be reported, it will already have been reported by another step.
+                    return;
+                }
+            }
+            else if (constructorArgument.Kind == TypedConstantKind.Array)
+            {
+                bool hadError = false;
+                parameters = ArrayBuilder<ParameterSymbol?>.GetInstance(constructorArgument.Values.Length);
+                var ordinalsBuilder = ArrayBuilder<int>.GetInstance(constructorArgument.Values.Length);
+                foreach (var nestedArgument in constructorArgument.Values)
+                {
+                    if (decodeName(nestedArgument, ref arguments) is (int ordinal, var parameter) && !hadError)
+                    {
+                        parameters.Add(parameter);
+                        ordinalsBuilder.Add(ordinal);
+                    }
+                    else
+                    {
+                        hadError = true;
+                    }
+                }
+
+                if (hadError)
+                {
+                    parameters.Free();
+                    ordinalsBuilder.Free();
+                    return;
+                }
+
+                parameterOrdinals = ordinalsBuilder.ToImmutableAndFree();
+            }
+            else
+            {
+                // There was some error already reported for having an incorrect argument.
+                return;
+            }
+
+            // Validate that the type has a constructor with the correct types.
+            var analyzedArguments = AnalyzedArguments.GetInstance();
+            var intType = this.DeclaringCompilation.GetSpecialType(SpecialType.System_Int32);
+            var intPlaceholder = new BoundInterpolatedStringArgumentPlaceholder(arguments.AttributeSyntaxOpt!, intType, hasErrors: intType.IsErrorType());
+            // literalLength
+            analyzedArguments.Arguments.Add(intPlaceholder);
+            analyzedArguments.RefKinds.Add(RefKind.None);
+            // formattedCount
+            analyzedArguments.Arguments.Add(intPlaceholder);
+            analyzedArguments.RefKinds.Add(RefKind.None);
+
+            foreach (var parameter in parameters)
+            {
+                var argumentType = parameter?.Type ?? ContainingType;
+                analyzedArguments.Arguments.Add(new BoundInterpolatedStringArgumentPlaceholder(arguments.AttributeSyntaxOpt!, argumentType, hasErrors: argumentType.IsErrorType()));
+                analyzedArguments.RefKinds.Add(parameter?.RefKind ?? RefKind.None);
+            }
+
+            // First try without an out parameter
+            bool success = binder.TryPerformConstructorOverloadResolution(
+                handlerType,
+                analyzedArguments,
+                errorName: handlerType.Name,
+                errorLocation,
+                suppressResultDiagnostics: true,
+                BindingDiagnosticBag.Discarded,
```

Comment:
>BindingDiagnosticBag.Discarded [](http://example.com/codeflow?start=16&length=30)

When I see us binding with ```BindingDiagnosticBag.Discarded``` and then reporting a diagnostics based on the result, that is a red flag for me. Used asembly references are not tracked and it is quite possible that things break after they are removed because compiler says they are not used.